core-bundle/src/Controller/ContentElement/PlayerController.php

```diff
@@ -113,11 +113,48 @@ function (FilesystemItem $item) use (&$captions, $range): HtmlAttributes {
             $sourceFiles,
         );
 
+        $tracks = [];
+        $textTrackFiles = [];
+
+        $setDefault = false;
+
+        if ($model->addTextTracks) {
+            $textTrackItems = FilesystemUtil::listContentsFromSerialized($this->filesStorage, $model->textTrackSRC ?: '');
+            $textTrackFiles = $this->getSourceFiles($textTrackItems);
+        }
+
+        if ([] !== $textTrackFiles) {
+            foreach ($textTrackFiles as $file) {
+                $textTrack = $file->getExtraMetadata()['textTrack'] ?? null;
+                $label = ($file->getExtraMetadata()['metadata'] ?? null)?->getDefault()?->getTitle();
+
+                if (empty($label) || !$textTrack?->getSourceLanguage()) {
+                    continue;
+                }
+
+                $trackAttributes = (new HtmlAttributes())
+                    ->setIfExists('kind', $textTrack->getType()?->value)
+                    ->set('label', $label)
+                    ->set('srclang', $textTrack->getSourceLanguage())
+                    ->set('src', $this->publicUriByStoragePath[$file->getPath()])
+                ;
+
+                // Set the first file as the default track
+                if (!$setDefault) {
+                    $trackAttributes->set('default');
+                    $setDefault = true;
+                }
+
+                $tracks[] = $trackAttributes;
+            }
+        }
+
```

Comment:
I would clean up things in this section as follows, most importantly we should not reuse `getSourceFiles()` as it also sorts things and stuff.

```suggestion
        $tracks = [];

        if ($model->addTextTracks) {
            $trackItems = FilesystemUtil::listContentsFromSerialized($this->filesStorage, $model->textTrackSRC ?: '');

            foreach ($trackItems as $trackItem) {
                if (!$publicUri = $this->filesStorage->generatePublicUri($trackItem->getPath())) {
                    continue;
                }

                $extraMetadata = $trackItem->getExtraMetadata();

                /** @todo change to: … = $extraMetadata->getTextTrack(); */
                if (null === ($textTrack = $extraMetadata['textTrack'] ?? null)) {
                    continue;
                }

                /** @todo change to: … = $extraMetadata->getLocalized()?->getDefault()?->getTitle(); */
                if (null === ($label = ($extraMetadata['metadata'] ?? null)?->getDefault()?->getTitle())) {
                    continue;
                }

                $tracks[] = (new HtmlAttributes())
                    ->setIfExists('kind', $textTrack->getType()?->value)
                    ->set('label', $label)
                    ->set('srclang', $textTrack->getSourceLanguage())
                    ->set('src', $publicUri)
                ;
            }

            // Set the first file as the default track
            ($tracks[0] ?? null)?->set('default');
        }

```